bait-files/src/test/java/com/google/sps/LoginTest.java

```diff
@@ -0,0 +1,59 @@
+package com.google.sps;
+
+import static com.google.appengine.api.datastore.FetchOptions.Builder.withLimit;
+import static com.google.sps.data.Keys.*;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.*;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.After;
+import org.junit.Test;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import java.io.*;
+import javax.servlet.http.*;
+import javax.servlet.ServletException;
+import com.google.sps.servlets.UsernameTakenServlet;
+import com.google.sps.servlets.LoginServlet;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class LoginTest {
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalUserServiceTestConfig())
+          .setEnvIsAdmin(true).setEnvIsLoggedIn(true).setEnvEmail("andrew").setEnvAuthDomain("gmail.com");
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  // Add a user and ensure there is a new entry in database
+  @Test
+  public void doPost_addsUserSuccessfully_returnsUserInDatabase() throws IOException, ServletException {
+        HttpServletRequest request = mock(HttpServletRequest.class);       
```

Comment:
Naming nit: the second part of the test name is meant to describe the setup or input of the particular test case.  "addsUserSuccessfully" sounds like a result... i.e. something that should be in the last part.

Maybe something like:
doPost_forNewUser_addsUserToDatabase

Or like I said before, if it feels overly obvious to include the middle part (the input / scenario), you could just have the first and last parts:
doPost_addsUserToDatabase

Although having said all that, when I look at the content of the actual doPost method, I see that the last thing it does is to apply a redirect on the response, so it could actually be good to test that.  You do have a mock of the response object in this test.  I forget the syntax for this, but from looking at Mockito documentation (https://github.com/mockito/mockito/wiki#examples) it could be something like:
verify(response).sendRedirect("/profile.jsp");

Then your test could be called something like:
doPost_addsUserAndRedirects or
doPost_forNewUser_addsUserAndRedirects

Reply:
Renamed tests